diff --git a/node_modules/vite/dist/client/client.mjs b/node_modules/vite/dist/client/client.mjs
index 20a879a..b23eb9b 100644
--- a/node_modules/vite/dist/client/client.mjs
+++ b/node_modules/vite/dist/client/client.mjs
@@ -546,8 +546,13 @@ function setupWebSocket(protocol, hostAndPath, onCloseWithoutOpen) {
   socket2.addEventListener("message", async ({ data }) => {
     handleMessage(JSON.parse(data));
   });
-  socket2.addEventListener("close", async ({ wasClean }) => {
-    if (wasClean) return;
+  let willUnload = false
+  window.addEventListener("beforeunload", () => {
+    willUnload = true
+  }, { once: true })
+  socket2.addEventListener("close", async () => {
+    // ignore close caused by top-level navigation
+    if (willUnload) return
     if (!isOpened && onCloseWithoutOpen) {
       onCloseWithoutOpen();
       return;
